tests/autorag/evaluate/metric/test_retrieval_metric.py

```diff
@@ -42,3 +42,10 @@ def test_retrieval_precision():
     result = retrieval_precision(retrieval_gt=retrieval_gt, pred_ids=pred)
     for gt, res in zip(solution, result):
         assert gt == pytest.approx(res, rel=1e-4)
+
+
+def test_retrieval_ndcg():
+    solution = [0.9197207891481876, 1, 1, 1, 1, None, None]
```

Comment:
In second case.
The IDCG will be 1 + (1/log_2 3) + 0.5 + 0
DCG will be 1 + 0 + 0 + 0 = 1
Thus, NDCG never be `1` 
I think your code is wrong. The solution in this test code does not make sense.

https://data-scient2st.tistory.com/193